### DISCUSSION TITLE
Include strategy matrix property based on existing matrix nested object

### DISCUSSION THREAD

**Discussion by markusSimonsen** [community/community#26779] [Actions]
<p>I have a general list of objects which I would like to reuse across environments. This list of objects is used as the strategy matrix in e.g. a deploy-development job.</p>
<pre><code class="lang-auto">[
  {
    "name": "name-1"
  },
  {
    "name": "name-2"
  },
  {
    "name": "name-3"
  },
  {
    "name": "name-4"
  },
  {
    "name": "name-5"
  }
]
</code></pre>
<p>In the deploy-development job, I need to have a property added to the matrix which is dependent on the entry of my matrix’s current value.<br>
(note that the matrix is being build and passed into this step from a previous step, so I can reuse the same matrix across jobs)</p>
<pre><code class="lang-auto">deploy-development:
    runs-on: ubuntu-latest
    if: github.event_name == 'push'
    needs: matrix-builder
    strategy:
      fail-fast: false
      matrix:
        matrix-entries: ${{ fromJson(needs.matrix-builder.outputs.matrix) }}
        include:
        - matrix-entries: { name: name-1 }
          new_property: new_property_value
    environment: development
    steps:
    - name: Checkout repository
      uses: actions/checkout@v2
</code></pre>
<p>What I’m trying to accomplish here is to add a new property (new_property) with a given value (new_property_value), when my matrix entry’s name is name-1.</p>
<p>The reason I need this functionality, is because there is values which is specific for the matrix entry and the environment combined.</p>

**Comment by markusSimonsen** [comment, marked as answer]
<p>Surprise surprise, I actually didn’t check this current way of doing it, and it actually works ^^’.<br>
I think the include checks if the matrix entry has a property called name, and if it’s the same value. If so, the new_property is added.</p>
<p>If anyone was wondering about sharing the same matrix across jobs, I used <a href="https://github.community/t/how-to-share-matrix-between-jobs/128595/8">this</a> way of doing it.</p>
<p>If that link somehow dies in the future, here is the job that does it:</p>
<pre><code class="lang-auto">matrix-builder:
    runs-on: ubuntu-latest
    if: github.event_name == 'push'
    outputs:
      matrix: ${{ steps.set-matrix.outputs.matrix }}
    steps:
    - uses: actions/checkout@v2
    - name: Assign the matrix from matrix.json
      id: set-matrix
      run: |
        TASKS=$(echo $(cat .github/workflows/domain-matrix.json) | sed 's/ //g' )
        echo "::set-output name=matrix::$TASKS"
</code></pre>
<p>It retrieves the json file and appends it as an output variable. Whenever you need to consume it, run ${{ fromJson(needs.matrix-builder.outputs.matrix }} as seen in the original post.</p>

**Comment by Simran-B** [comment]
<p>FWIW, the documentation has an example of sharing matrices between jobs: <a href="https://docs.github.com/en/actions/learn-github-actions/expressions#example-returning-a-json-object" class="inline-onebox" rel="noopener nofollow ugc">Expressions - GitHub Docs</a></p>
